auth/auth0/src/auth0Authentication.js

```diff
@@ -0,0 +1,104 @@
+const isomorphicFetch = require('isomorphic-fetch');
+const jwt = require('jsonwebtoken');
+const jwkRsa = require('jwks-rsa');
+const fromEvent = require('graphcool-lib').fromEvent;
+
+//Validates the request JWT token
+const verifyToken = token =>
+  new Promise(resolve => {
+    //Decode the JWT Token
+    const decoded = jwt.decode(token, { complete: true });
+    if (!decoded || !decoded.header || !decoded.header.kid) {
+      throw new Error('Unable to retrieve key identifier from token');
+    }
+    if (decoded.header.alg !== 'RS256') {
+      throw new Error(
+        `Wrong signature algorithm, expected RS256, got ${decoded.header.alg}`
+      );
+    }
+    const jkwsClient = jwkRsa({
+      cache: true,
+      jwksUri: `https://${process.env.AUTH0_DOMAIN}/.well-known/jwks.json`
+    });
+    //Retrieve the JKWS's signing key using the decode token's key identifier (kid)
+    jkwsClient.getSigningKey(decoded.header.kid, (err, key) => {
+      if (err) throw new Error(err);
+      const signingKey = key.publicKey || key.rsaPublicKey;
+      //If the JWT Token was valid, verify its validity against the JKWS's signing key
+      jwt.verify(
+        token,
+        signingKey,
+        {
+          algorithms: ['RS256'],
+          audience: process.env.AUTH0_API_IDENTIFIER,
+          ignoreExpiration: false,
+          issuer: `https://${process.env.AUTH0_DOMAIN}/`
+        },
+        (err, decoded) => {
+          if (err) throw new Error(err);
+          return resolve(decoded);
+        }
+      );
+    });
+  });
+
+//Retrieves the Graphcool user record using the Auth0 user id
+const getGraphcoolUser = (auth0UserId, api) =>
+  api
+    .request(
+      `
+        query getUser($auth0UserId: String!){
+          User(auth0UserId: $auth0UserId){
+            id
+          }
+        }
+      `,
+      { auth0UserId }
+    )
+    .then(queryResult => queryResult.User);
+
+//Creates a new User record.
+const createGraphCoolUser = ({ sub }, api) =>
+  api
+    .request(
+      `
+        mutation createUser($auth0UserId: String!) {
+          createUser(
+            auth0UserId: $auth0UserId
+          ){
+            id
+          }
+        }
+      `,
+      { auth0UserId: sub }
+    )
+    .then(queryResult => queryResult.createUser);
+
+export default async event => {
```

Comment:
Don't you need `'use latest'` to use `async`?

Reply:
Not anymore. The new function runtime is on AWS. `'use latest'` was a Auth0 Extend (webtask) specific instruction.